Update OmniDiskSweeper Recipes to remove the com.apple.FinderInfo attribute.

paul-cossey opened the pull request:
Hi, Folks 

The OmniDiskSweeper recipes are currently failing with the below error

```
Processor: CodeSignatureVerifier: Error: Code signature verification failed. Note that all verifications can be disabled by setting the variable DISABLE_CODE_SIGNATURE_VERIFICATION to a non-empty value.
```

After some detective work by BigMacAdmin and elios in Mac Admins Slack they figured out it was the vendor (mistakenly?) including the `com.apple.FinderInfo` attribute to the app bundle. 

This PR adds in steps to copy the bundle out of the dmg remove the attribute and create a new dmg. Please note this requires using a custom `QuarantineRemover` processor available in the [dataJAR Shared Processor](https://github.com/jamf/private-recipes/tree/master/Shared%20Processors) folder. 

Output from a successful -v munki run 
```
autopkg run -v OmniDiskSweeper.munki.recipe   
Looking for com.github.autopkg.download.OmniDiskSweeper...
Did not find com.github.autopkg.download.OmniDiskSweeper in recipe map
Rebuilding recipe map with current working directories...
Looking for com.github.autopkg.download.OmniDiskSweeper...
Found com.github.autopkg.download.OmniDiskSweeper in recipe map
Looking for com.github.autopkg.download.omnigroupproduct...
Found com.github.autopkg.download.omnigroupproduct in recipe map
**load_recipe time: 0.01805491700361017
Processing OmniDiskSweeper.munki.recipe...
WARNING: OmniDiskSweeper.munki.recipe is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
SparkleUpdateInfoProvider
SparkleUpdateInfoProvider: Items in feed: 7
SparkleUpdateInfoProvider: Items in default channel: 7
SparkleUpdateInfoProvider: Version retrieved from appcast: 1.16
SparkleUpdateInfoProvider: Found URL https://downloads.omnigroup.com/software/macOS/11/OmniDiskSweeper-1.16.dmg
URLDownloader
URLDownloader: Storing new Last-Modified header: Fri, 12 Sep 2025 22:51:15 GMT
URLDownloader: Storing new ETag header: "80b92e-63ea27dbd57b4"
URLDownloader: Downloaded /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/downloads/OmniDiskSweeper-1.16.dmg
EndOfCheckPhase
Copier
Copier: Mounted disk image /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/downloads/OmniDiskSweeper-1.16.dmg
Copier: Copied /private/tmp/dmg.HASAsF/OmniDiskSweeper.app to /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/Applications/OmniDiskSweeper.app
com.github.dataJAR-recipes.Shared Processors/QuarantineRemover
DmgCreator
DmgCreator: Created dmg from /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/Applications at /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/OmniDiskSweeper.dmg
CodeSignatureVerifier
CodeSignatureVerifier: Mounted disk image /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/OmniDiskSweeper.dmg
CodeSignatureVerifier: Verifying code signature...
CodeSignatureVerifier: Deep verification enabled...
CodeSignatureVerifier: Strict verification enabled...
CodeSignatureVerifier: /private/tmp/dmg.K8y2x9/OmniDiskSweeper.app: valid on disk
CodeSignatureVerifier: /private/tmp/dmg.K8y2x9/OmniDiskSweeper.app: satisfies its Designated Requirement
CodeSignatureVerifier: /private/tmp/dmg.K8y2x9/OmniDiskSweeper.app: explicit requirement satisfied
CodeSignatureVerifier: Signature is valid
MunkiImporter
MunkiImporter: Using repo lib: AutoPkgLib
MunkiImporter:         plugin: FileRepo
MunkiImporter:           repo: /Users/Shared/munki_repo
MunkiImporter: Copied pkginfo to: /Users/Shared/munki_repo/pkgsinfo/apps/OmniDiskSweeper-1.16.plist
MunkiImporter:            pkg to: /Users/Shared/munki_repo/pkgs/apps/OmniDiskSweeper-1.16.dmg
Receipt written to /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/receipts/OmniDiskSweeper.munki-receipt-20250918-142458.plist

The following new items were downloaded:
    Download Path                                                                                                         
    -------------                                                                                                         
    /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.munki.OmniDiskSweeper/downloads/OmniDiskSweeper-1.16.dmg  

The following new items were imported into Munki:
    Name             Version  Catalogs  Pkginfo Path                     Pkg Repo Path                  Icon Repo Path  
    ----             -------  --------  ------------                     -------------                  --------------  
    OmniDiskSweeper  1.16     testing   apps/OmniDiskSweeper-1.16.plist  apps/OmniDiskSweeper-1.16.dmg
```

paul-cossey committed:
Update OmniDiskSweeper Recipes

Updated with steps to work around the vendor including the `com.apple.FinderInfo` attribute
```
Processor: CodeSignatureVerifier: Error: Code signature verification failed. Note that all verifications can be disabled by setting the variable DISABLE_CODE_SIGNATURE_VERIFICATION to a non-empty value.
```

gregneagle commented:
I think it would be extremely confusing for new AutoPkg users to have to add the dataJAR recipes in order to use a recipe from the recipes repo. 
I don’t have a strong suggestion here on what to do instead, though.

gregneagle commented:
Since the existing Munki recipe expects the app is in a disk image, yes, putting it back into a disk image is a reasonable thing to do here. I think the pkg recipe also expects the app to be in a disk image...

nmcspadden commented:
I agree with @gregneagle that I don't want the recipes repo to depend on another repo.

So that means our options are:
1) We deprecate this recipe out of the core recipes repo and move everyone to DataJar's;
2) We bring in the shared processor into core (which I don't really want to do)
3) We accept this as a temporary requirement until the vendor fixes it (assuming they do); if this persists longer than the next release (i.e. appears permanent), then we have to re-consider 1-2.

paul-cossey commented:
Or option 4. I amend the PR to remove 
```
<key>strict_verification</key>
<true/>
```

Looks like it's not a key that's used in most recipes (148) https://github.com/search?q=org%3Aautopkg%20%3Ckey%3Estrict_verification%3C%2Fkey%3E%20%3Ctrue%2F%3E&type=code 

The recipe will then complete 
```
autopkg run -vv OmniDiskSweeper.download.recipe
Looking for com.github.autopkg.download.omnigroupproduct...
Did not find com.github.autopkg.download.omnigroupproduct in recipe map
Rebuilding recipe map with current working directories...
Looking for com.github.autopkg.download.omnigroupproduct...
Found com.github.autopkg.download.omnigroupproduct in recipe map
**load_recipe time: 0.021348833000047307
Processing OmniDiskSweeper.download.recipe...
WARNING: OmniDiskSweeper.download.recipe is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
SparkleUpdateInfoProvider
{'Input': {'alternate_xmlns_url': 'https://www.omnigroup.com/namespace/omniappcast/v1',
           'appcast_url': 'https://update.omnigroup.com/appcast/com.omnigroup.OmniDiskSweeper'}}
SparkleUpdateInfoProvider: Items in feed: 7
SparkleUpdateInfoProvider: Items in default channel: 7
SparkleUpdateInfoProvider: Version retrieved from appcast: 1.16
SparkleUpdateInfoProvider: Found URL https://downloads.omnigroup.com/software/macOS/11/OmniDiskSweeper-1.16.dmg
{'Output': {'url': 'https://downloads.omnigroup.com/software/macOS/11/OmniDiskSweeper-1.16.dmg',
            'version': '1.16'}}
URLDownloader
{'Input': {'url': 'https://downloads.omnigroup.com/software/macOS/11/OmniDiskSweeper-1.16.dmg'}}
URLDownloader: No value supplied for prefetch_filename, setting default value of: False
URLDownloader: No value supplied for CHECK_FILESIZE_ONLY, setting default value of: False
URLDownloader: Item at URL is unchanged.
URLDownloader: Using existing /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.download.OmniDiskSweeper/downloads/OmniDiskSweeper-1.16.dmg
{'Output': {'pathname': '/Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.download.OmniDiskSweeper/downloads/OmniDiskSweeper-1.16.dmg'}}
EndOfCheckPhase
{'Input': {}}
{'Output': {}}
CodeSignatureVerifier
{'Input': {'input_path': '/Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.download.OmniDiskSweeper/downloads/OmniDiskSweeper-1.16.dmg/OmniDiskSweeper.app',
           'requirement': 'identifier "com.omnigroup.OmniDiskSweeper" and '
                          'anchor apple generic and certificate '
                          '1[field.1.2.840.113635.100.6.2.6] /* exists */ and '
                          'certificate leaf[field.1.2.840.113635.100.6.1.13] '
                          '/* exists */ and certificate leaf[subject.OU] = '
                          '"34YW5XSRB7"'}}
CodeSignatureVerifier: Mounted disk image /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.download.OmniDiskSweeper/downloads/OmniDiskSweeper-1.16.dmg
CodeSignatureVerifier: Verifying code signature...
CodeSignatureVerifier: Deep verification enabled...
CodeSignatureVerifier: Strict verification not defined. Using codesign defaults...
CodeSignatureVerifier: /private/tmp/dmg.nhn0KP/OmniDiskSweeper.app: valid on disk
CodeSignatureVerifier: /private/tmp/dmg.nhn0KP/OmniDiskSweeper.app: satisfies its Designated Requirement
CodeSignatureVerifier: /private/tmp/dmg.nhn0KP/OmniDiskSweeper.app: explicit requirement satisfied
CodeSignatureVerifier: Signature is valid
{'Output': {}}
Receipt written to /Users/paul.cossey/Library/AutoPkg/Cache/com.github.autopkg.download.OmniDiskSweeper/receipts/OmniDiskSweeper.download-receipt-20250918-162858.plist

Nothing downloaded, packaged or imported.
```

paul-cossey committed:
Update OmniDiskSweeper to remove strict_verification

Updated the PR to remove strict_verification from CodeSignatureVerifier and remove unnecessary steps.

Reverts back to using pathname.

paul-cossey commented:
Sorry I didn't get round to this Friday (my laptop needed an erase and reinstall) 

I've just amended to go with option 4 and remove strict verification https://github.com/autopkg/recipes/pull/507/commits/eaf5e9633091af14e2ff3792575501c40762314a
